GoogleForCreators/web-stories-wp
# Flush rewrite rules on plugin activation

## Summary

(Soft) Flushes rewrite rules on plugin activation and deactivation.

## Relevant Technical Choices

Ignore WP VIP specific PHPCS rule that seem to be built more for themes anyway

## To-do

N/A

## User-facing changes

N/A

## Testing Instructions

Test plugin activation & permalinks on a clean site. Use `wp rewrite` to check rewrite rules.

Rewrite rules **before** plugin **activation**:

```
$ wp rewrite list --match=stories/ffoo
+--------------------------+-------------------------------------------------+--------+
| match                    | query                                           | source |
+--------------------------+-------------------------------------------------+--------+
| (.?.+?)(?:/([0-9]+))?/?$ | index.php?pagename=$matches[1]&page=$matches[2] | page   |
| [^/]+/([^/]+)/?$         | index.php?attachment=$matches[1]                | post   |
+--------------------------+-------------------------------------------------+--------+
```

Rewrite  rules **after** plugin **activation**:

```
$ wp rewrite list --match=stories/ffoo
+----------------------------------+--------------------------------------------------+-----------+
| match                            | query                                            | source    |
+----------------------------------+--------------------------------------------------+-----------+
| stories/([^/]+)(?:/([0-9]+))?/?$ | index.php?web-story=$matches[1]&page=$matches[2] | web-story |
| (.?.+?)(?:/([0-9]+))?/?$         | index.php?pagename=$matches[1]&page=$matches[2]  | page      |
| [^/]+/([^/]+)/?$                 | index.php?attachment=$matches[1]                 | post      |
+----------------------------------+--------------------------------------------------+-----------+
```

Rewrite rules **after** plugin **deactivation**:

```
$ wp rewrite list --match=stories/ffoo
+--------------------------+-------------------------------------------------+--------+
| match                    | query                                           | source |
+--------------------------+-------------------------------------------------+--------+
| (.?.+?)(?:/([0-9]+))?/?$ | index.php?pagename=$matches[1]&page=$matches[2] | page   |
| [^/]+/([^/]+)/?$         | index.php?attachment=$matches[1]                | post   |
+--------------------------+-------------------------------------------------+--------+
```

----

Fixes #2174
